daemons/gptp/common/common_port.cpp

```diff
@@ -487,57 +497,38 @@ bool CommonPort::processSyncAnnounceTimeout( Event e )
 			   1000000000.0)));
 	}
 
-	if ( getPortState() == PTP_INITIALIZING ||
-	     getPortState() == PTP_UNCALIBRATED ||
-	     getPortState() == PTP_SLAVE ||
-	     getPortState() == PTP_PRE_MASTER )
-	{
-		GPTP_LOG_STATUS(
-			"*** %s Timeout Expired - Becoming Master",
-			e == ANNOUNCE_RECEIPT_TIMEOUT_EXPIRES ? "Announce" :
-			"Sync" );
-		{
-			// We're Grandmaster, set grandmaster info to me
-			ClockIdentity clock_identity;
-			unsigned char priority1;
-			unsigned char priority2;
-			ClockQuality clock_quality;
-
-			clock_identity = getClock()->getClockIdentity();
-			getClock()->setGrandmasterClockIdentity
-				( clock_identity );
-			priority1 = getClock()->getPriority1();
-			getClock()->setGrandmasterPriority1( priority1 );
-			priority2 = getClock()->getPriority2();
-			getClock()->setGrandmasterPriority2( priority2 );
-			clock_quality = getClock()->getClockQuality();
-			getClock()->setGrandmasterClockQuality
-				( clock_quality );
-		}
-		setPortState( PTP_MASTER );
-		Timestamp system_time;
-		Timestamp device_time;
+	if( getPortState() == PTP_MASTER )
+		return true;
 
```

Comment:
When we are already GM, we should be sending SYNC and ANNOUNCE, but I think we'll only receive them if someone thinks they're higher priority and wants to take over.  I guess there might be a leftover timer from a recent change in status that could get us here?  Probably good to guard against the situation even if unlikely.